src/edid.c

```diff
@@ -472,20 +464,30 @@ static void decode_detailed_timing(const unsigned char *timing, struct detailed_
 		FOUR_WAY_INTERLEAVED, SIDE_BY_SIDE
 	};
 
-	detailed->pixel_clock = (timing[0x00] | timing[0x01] << 8) * 10000;
-	detailed->h_addr = timing[0x02] | ((timing[0x04] & 0xf0) << 4);
-	detailed->h_blank = timing[0x03] | ((timing[0x04] & 0x0f) << 8);
-	detailed->v_addr = timing[0x05] | ((timing[0x07] & 0xf0) << 4);
-	detailed->v_blank = timing[0x06] | ((timing[0x07] & 0x0f) << 8);
-	detailed->h_front_porch = timing[0x08] | get_bits(timing[0x0b], 6, 7) << 8;
-	detailed->h_sync = timing[0x09] | get_bits(timing[0x0b], 4, 5) << 8;
-	detailed->v_front_porch = get_bits(timing[0x0a], 4, 7) | get_bits(timing[0x0b], 2, 3) << 4;
-	detailed->v_sync = get_bits(timing[0x0a], 0, 3) | get_bits(timing[0x0b], 0, 1) << 4;
-	detailed->width_mm = timing[0x0c] | get_bits(timing[0x0e], 4, 7) << 8;
-	detailed->height_mm = timing[0x0d] | get_bits(timing[0x0e], 0, 3) << 8;
+	detailed->pixel_clock = (timing[0x00] |
```

Comment:
Sorry, I realize now I didn't make much sense.  I meant something iike this:

```c
	detailed->pixel_clock = (timing[0x00] |   timing[0x01] << 8) * 10000;
	detailed->h_addr      =  timing[0x02] | ((timing[0x04] & 0xf0) << 4);
	detailed->h_blank     =  timing[0x03] | ((timing[0x04] & 0x0f) << 8);
	detailed->v_addr      =  timing[0x05] | ((timing[0x07] & 0xf0) << 4);
	detailed->v_blank     =  timing[0x06] | ((timing[0x07] & 0x0f) << 8);
```

Now, after aligning the code around the `=` sign, my eye immediately goes to the first line: why isn't that masked with `0x0f`?  There's probably more, but that's just an example of what I wanted to point out :)

Reply:
Oh now that's even better readability. Truth be told, I've just taken code from some sources I tried a few months back and have not dug into it really. Just know that it works good enough for now. I am sure there'll be issues with parsing later on though...